src/v/storage/storage_resources.h

```diff
@@ -42,7 +43,9 @@ namespace storage {
 class adjustable_allowance {
 public:
     adjustable_allowance(uint64_t capacity)
-      : _sem(capacity)
+      : adjustable_allowance(capacity, "s-allowance") {}
+    adjustable_allowance(uint64_t capacity, const ss::sstring& sem_name)
+      : _sem(ssx::make_named_semaphore(capacity, sem_name))
```

Comment:
Is it super painful to make this always named?

Reply:
Yeah I had a TODO to that effect in the ssx include file. This meets a more immediate need to get a handle on CI failures, so it seems like we should get something merged here and then I can submit a patch to seastar, and bring back any changes they suggest. What do you think?